lib/Dialect/TTNN/Utils/TransformUtils.cpp

```diff
@@ -31,4 +32,60 @@ Value getOrInsertDevice(PatternRewriter &rewriter, Operation *op) {
   rewriter.restoreInsertionPoint(currentInsertionPoint);
   return deviceOp.getResult();
 }
+
+// Helper method to insert a ToLayoutOp to convert the input operand to the
+// desired tensor layout, buffer type and memory layout.
+mlir::Value
+createToLayoutOp(Operation *op, Value inputValue, PatternRewriter &rewriter,
+                 Layout targetTensorLayout, BufferType targetTensorBufferType,
+                 std::optional<TensorMemoryLayout> targetTensorMemoryLayout) {
+  TTNNLayoutAttr inputLayoutAttr = getLayoutAttrFromValue(inputValue);
+
+  // Create element type based on tensor layout.
+  Type elementType = getElementType(rewriter.getContext(), targetTensorLayout,
+                                    inputLayoutAttr.getDataType());
+
+  // Create tensor memory layout attribute.
+  ttnn::TensorMemoryLayoutAttr outputMemLayoutAttr =
+      targetTensorMemoryLayout.has_value()
+          ? ttnn::TensorMemoryLayoutAttr::get(rewriter.getContext(),
+                                              targetTensorMemoryLayout.value())
+          : nullptr;
+
+  // Create the output memory config attribute.
+  ttnn::MemoryConfigAttr outputMemConfigAttr = ttnn::MemoryConfigAttr::get(
+      rewriter.getContext(),
+      ttnn::BufferTypeAttr::get(rewriter.getContext(), targetTensorBufferType),
+      ttnn::ShardSpecAttr::get(
+          op->getContext(),
+          ttnn::ShapeAttr::get(rewriter.getContext(),
+                               inputLayoutAttr.getMemref().getShape())),
+      outputMemLayoutAttr);
+
+  // Get the input operand type.
+  RankedTensorType inputOperandType =
+      mlir::cast<RankedTensorType>(inputValue.getType());
+
+  // Create a ToLayoutOp to convert the input operand to the desired
+  // tensor layout, buffer type and memory layout.
+  return rewriter
+      .create<ttnn::ToLayoutOp>(
+          op->getLoc(),
+          ttnn::utils::createRankedTensorTypeWithEncoding(
+              inputOperandType,
```

Comment:
If we apply data type workaround for example from f32 -> bf16 inputOperandType here would be tensor<32x32xbf16>, but here you want to create new tensor with outputType i.e tensor<32x32xf32>

Reply:
Actually nvm, you don't have data type workaround yet. This is fine

Reply:
Not relevant for now, but noted!